python/orca/example/learn/pytorch/resnet50/inference.py

```diff
@@ -99,87 +92,67 @@ def validate(self, val_iterator, info, metrics, num_steps=None):
                 weight=PerChannelMinMaxObserver.with_args(dtype=torch.qint8,
                                                           qscheme=torch.per_channel_symmetric))
             x = torch.randn(1, 3, 224, 224)
-            prepared_model = ipex.quantization.prepare(self.model, qconfig, x, inplace=True)
+            prepared_model = ipex.quantization.prepare(runner.model, qconfig, x, inplace=True)
             with torch.no_grad():
-                for i, (images, target) in enumerate(val_iterator):
+                for i, (images, target) in enumerate(runner.val_loader):
                     images = images.contiguous(memory_format=torch.channels_last)
                     prepared_model(images)
                     if i == 4:
                         print(i)
                         break
-                prepared_model.save_qconf_summary(self.config["configure_dir"])
+                prepared_model.save_qconf_summary(runner.config["configure_dir"])
                 print(".........calibration step done..........")
 
-        if not num_steps:
-            num_steps = len(val_iterator)
-        invalidInputError(num_steps > self.config["warmup_iterations"],
+        if not runner.num_steps:
+            runner.num_steps = len(runner.val_loader)
+        invalidInputError(runner.num_steps > runner.config["warmup_iterations"],
                           "total steps should be larger than warmup iterations")
+        if runner.config["warmup_iterations"] > 0:
+            print("running warmup iterations")
+
+    def on_val_forward(self, runner):
+        if runner.config["dummy"]:
+            images = torch.randn(runner.config["batch"], 3, 224, 224)
+            target = torch.arange(1, runner.config["batch"] + 1).long()
+            # Only do the conversion once for dummy data
+            if runner.config["ipex"]:
+                images = images.contiguous(memory_format=torch.channels_last)
+            if runner.config["bf16"]:
+                images = images.to(torch.bfloat16)

```

Comment:
This should be done only once but not for each iteration. Re-check the logic of the original code?